physionet-django/training/views.py

```diff
@@ -175,3 +249,103 @@ def download_course(request, training_slug, version):
     response = JsonResponse(response_data, safe=False, json_dumps_params={'indent': 2})
     response['Content-Disposition'] = f'attachment; filename={training_type.name}--version-{version}.json'
     return response
+
+
+@login_required
+def take_training(request, training_slug):
+
+    course = Course.objects.filter(training_type__slug=training_slug, is_active=True).order_by('version').last()
+    if course is None:
+        raise Http404()
+    modules = Module.objects.filter(course=course).order_by('order')
+    # get the progress of the user for the modules, updated_date
+    course_progress, _ = CourseProgress.objects.get_or_create(user=request.user, course=course)
+
+    for module in modules:
+        module_progress = course_progress.module_progresses.filter(module_id=module.id).last()
+        if module_progress:
+            module.progress_status = module_progress.get_status_display()
+            module.progress_updated_date = module_progress.updated_at
+            # if the module is completed, then put the last completed order as 1
+            # Else put it as the last_completed_order
+            if module_progress.status == ModuleProgress.Status.COMPLETED:
+                module.last_completed_order = 1
+            else:
+                module.last_completed_order = module_progress.last_completed_order
+        else:
+            module.progress_status = 'Not Started'
+            module.progress_updated_date = None
+            module.last_completed_order = 1
+
+    return render(request, 'training/course.html', {
+        'course': course,
+        'modules': modules,
+        'ModuleStatus': ModuleProgress.Status,
+    })
+
+
+@login_required
+def current_module_block(request, training_slug, module_order, order):
+
```

Comment:
This function should have a docstring, at minimum explaining its purpose. I think we also want to try to work towards simpler functions at some point, perhaps moving logic to a separate module.